### DISCUSSION TITLE
Computed property dependency injection

### DISCUSSION THREAD

**Discussion by gbuckingham89** [livewire/livewire#6967] [4. General]
I'm just upgrading a Laravel app that has several Livewire components from Livewire v2 to Livewire  v3 and have stumbled into a problem with dependency injection with computed properties. 

In v2, it was possible to inject dependencies into the computed property methods;

```php
public function getFooProperty(Bar $bar) { }
```

In v3, the recommended way to define computed properties is now via attributes;

```php
#[Computed]
public function foo() { }
```

If I define an argument that is resolvable via the container, I get an exception;

```php
#[Computed]
public function foo(Bar $bar) { }
```

```
ArgumentCountError

Too few arguments to function App\Http\Livewire\MyComponent::foo(), 0 passed and exactly 1 expected
```

The v2 method of defining computed properties does still work in v3, but this means I'm ending up with a mix of old _(so I can keep the DI)_ and new _(following current documentation)_ style definitions - which is inconsistent.

_I haven't labelled this a "bug" because I can't see the use of DI in computed properties included in the documentation for v2 or v3._

**Comment by kirkbushell** [comment]
Surprising this isn't getting more votes, came across this also while migrating an application...

**Reply by luanfreitasdev** [reply]
I submitted a Pull Request

https://github.com/livewire/livewire/pull/7690

**Reply by Gupyzer0** [reply]
Find myself with the same problem, in the end did something like this ...

```php
class Foo extends Component
{
    public $my_list_of_stuff = [];
    
    #[Computed]
    public function computedFilteredListOfStuff()
    {
     .... // Filtered my_list_of_stuff here using input from a form
    }
    
    // Used repository to get list of stuff ...
    public function refreshListUsingRepository(MyRepository $my_repository)
    {
        $this->my_list_of_stuff = $my_repository->all();
    }
}
```
